### ISSUE TITLE
Show the token when starting a notebook from the REPL

### ISSUE THREAD

**Issue by vchuravy** [JuliaLang/IJulia.jl#586]
Would it be possible to show the token of a new Jupyter instance when doing `notebook()`?
When starting through a ssh tunnel a browser can't be automatically started and my current workaround is:

```
ssh -L 8888:localhost:8888 HOST
> julia
julia> using IJulia
julia> notebook(detached=true)
julia> run(`$(IJulia.notebook_cmd[1]) notebook list`)
# Get the token used by the instance bound to localhost:8888
# and visit localhost:8888 in your browser and past the token their
```

**Comment by stevengj** [comment]
`notebook` could also start the notebook with token authentication disabled...

**Comment by vchuravy** [comment]
Yes that would be lovely, but not easy see https://github.com/jupyter/notebook/issues/2254 
You apparently need to either set a password or set a configuration option to set the token to be empty.

**Comment by joachimneu** [comment]
Considering the way Jupyter is used in IJulia/Julia, this seems like a reasonable suggestion: "Would it be possible to show the token of a new Jupyter instance when doing notebook()?"

You can also consider setting up a password, along the lines of: https://discourse.julialang.org/t/ijulia-password-token-requirement/4974 (Careful, that "howto" is not completely accurate!)

@stevengj : You might be sharing the same machine with multiple users in a cluster environment. You would not want other users to be able to execute code with your user's privileges. Tokens prevent that.

**Comment by Moelf** [comment]
`notebook_cmd` is no longer working

**Comment by tpgillam** [comment]
This would be useful! As a workaround, I'm now using the following shameful hack with IJulia 1.23.2:

```julia
using IJulia
jupyter = IJulia.find_jupyter_subcommand("")[1]
run(`$(jupyter) notebook list`)
```

**Comment by glemieux** [comment]
> This would be useful! As a workaround, I'm now using the following shameful hack with IJulia 1.23.2:
> 
> ```julia
> using IJulia
> jupyter = IJulia.find_jupyter_subcommand("")[1]
> run(`$(jupyter) notebook list`)
> ```

I wanted to note that this helped me troubleshoot a problem during a clean install of conda and jupyter via IJulia.  Previous to the clean install I had been tinkering around with hosting jupyter on a remote server and one of the server links was still open and connected to the default url and ports.  As a result this caused a connection failure when trying to run the newly installed jupyter via IJulia.  Long story short, I was able to use this (without installing jupyter by some other means) to see all the notebooks lingering in the background that had failed to close down properly and were conflicting with this

**Comment by uajdgiz** [comment]
> This would be useful! As a workaround, I'm now using the following shameful hack with IJulia 1.23.2:
> 
> ```julia
> using IJulia
> jupyter = IJulia.find_jupyter_subcommand("")[1]
> run(`$(jupyter) notebook list`)
> ```

Thanks for this.  I am brand new to WSL and this is this only way I could get the notebook running in the Windows browser.  It's been a few years, but has there been anything implemented more permanent?  Or a better way to pass the token?

**Comment by nlitz88** [comment]
@uajdgiz I am also facing this issue.

Launching the notebook from the Julia REPL feels a little hacky--or just like a small convenience for most users accessing the notebook from their local machine. Not to say I wouldn't appreciate this feature being updated to eliminate the need for a token or print it out on launch, but yeah--it feels like there should be a better way.

The best thing I could think of was just running a Jupyterlab docker container with Julia built in. I found a helpful guide [here](https://www.docker.com/blog/supercharging-ai-ml-development-with-jupyterlab-and-docker/) to introduce the concept, and there's more documentation on [Jupyter's documentation](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/selecting.html#jupyter-julia-notebook) on doing this. This feels like the _right_ way to handle this.

Unfortunately for me, I need Julia 1.6.7, but their docker images only go back to 1.9.1 or something--so I wasn't able to personally take this route without making my own image. For now (and for the sake of time) I'll be sticking to the "shameful hack" lol.

**Comment by JamesWrigley** [comment]
I'm gonna call this fixed because since IJulia 1.28 one can pass `verbose=true` and thus see the token printed by Jupyter:
```
[C 2025-06-15 23:03:19.335 ServerApp] 
    
    To access the server, open this file in a browser:
        file:///home/james/.local/share/jupyter/runtime/jpserver-1504079-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/tree?token=b41d25350cec4777c486259fc99ef1a1f881531edf6ce78b
        http://127.0.0.1:8888/tree?token=b41d25350cec4777c486259fc99ef1a1f881531edf6ce78b
```

https://julialang.github.io/IJulia.jl/stable/changelog/#[v1.28.0](https://github.com/JuliaLang/IJulia.jl/releases/tag/v1.28.0)-2025-06-01
